kafkarator/connection_test.go

```diff
@@ -0,0 +1,198 @@
+package kafkarator
+
+import (
+	"bytes"
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hafslundkraft/golib/telemetry"
+	"github.com/segmentio/kafka-go"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	testkafka "github.com/testcontainers/testcontainers-go/modules/kafka"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+)
+
+const kafkaImage = "confluentinc/confluent-local:7.5.0"
+
+func Test_connection_Consumer(t *testing.T) {
+	ctx := context.Background()
+
+	// Set up the global text map propagator for W3C Trace Context
+	// This is required for trace context propagation to work
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(
+		propagation.TraceContext{},
+		propagation.Baggage{},
+	))
+
+	var buf bytes.Buffer
+
+	telClosed := false
+	tel, telClose := telemetry.New(
+		ctx,
+		"kafka-test",
+		telemetry.WithLocal(true),
+		telemetry.WithLocalWriter(&buf))
+	defer func() {
+		if telClosed {
+			return
+		}
+		telClose(ctx)
+	}()
+
+	// Start Kafka container
+	_, brokers, closer := startTestContainer(ctx, t)
+	defer closer()
+
+	ctx, span := tel.Tracer().Start(ctx, "test_connection")
+	defer span.End()
+
+	// Create a topic using kafka-go
+	topicName := "test-topic"
+	adminCloser := createTopic(t, topicName, brokers[0])
+	defer adminCloser()
+
+	// Create Config for New() - TLS is optional, so we can omit cert files for local testing
+	config := Config{
+		Brokers: brokers,
+	}
+
+	// Initialize connection
+	conn, err := New(config, tel)
+	if err != nil {
+		t.Fatalf("failed to create connection: %v", err)
+	}
```

Comment:
Hva med:
```suggestion
	require.NoError(t, err)
```
Og tilsvarende i andre error-checks i testen?